File: backend/app/schemas/admin_dashboard.py
"""Admin dashboard request/response schemas."""

from __future__ import annotations

import datetime

from pydantic import BaseModel


class TokenStats(BaseModel):
    """Token statistics."""

    input: int
    output: int


class DashboardSummaryResponse(BaseModel):
    """Response schema for dashboard summary."""

    total_users: int
    active_users: int
    total_conversations: int
    today_conversations: int
    total_messages: int
    total_tokens: TokenStats
    total_cost_usd: float


class TrendDataPoint(BaseModel):
    """Single data point for trend chart."""

    date: datetime.date
    value: int


class DashboardTrendsResponse(BaseModel):
    """Response schema for dashboard trends."""

    period: str
    metric: str
    data: list[TrendDataPoint]


class UserRankingItem(BaseModel):
    """Single user in ranking."""

    user_id: int
    email: str
    name: str | None
    value: int


class DashboardRankingsResponse(BaseModel):
    """Response schema for dashboard rankings."""

    metric: str
    rankings: list[UserRankingItem]


__all__ = [
    "TokenStats",
    "DashboardSummaryResponse",
    "TrendDataPoint",
    "DashboardTrendsResponse",
    "UserRankingItem",
    "DashboardRankingsResponse",
]

File: frontend/src/types/adminDashboard.ts
/**
 * Admin Dashboard types
 */

export interface TokenStats {
  input: number
  output: number
}

export interface DashboardSummary {
  total_users: number
  active_users: number
  total_conversations: number
  today_conversations: number
  total_messages: number
  total_tokens: TokenStats
  total_cost_usd: number
}

export interface TrendDataPoint {
  date: string
  value: number
}

export interface DashboardTrends {
  period: string
  metric: string
  data: TrendDataPoint[]
}

export interface UserRankingItem {
  user_id: number
  email: string
  name: string | null
  value: number
}

export interface DashboardRankings {
  metric: string
  rankings: UserRankingItem[]
}

export type TrendMetric = 'conversations' | 'messages' | 'tokens'
export type TrendPeriod = '7d' | '30d' | '90d' | 'custom'
export type RankingPeriod = '7d' | '30d' | '90d' | 'all'
